misc.c

```diff
@@ -1699,3 +1700,40 @@ x2dev(s, d)
 	*d = r;
 	return(s);
 }
+
+int
+compare_namespace(ns)
+	char *ns;                /* namespace */
+{
+	char nspath_self[100], nspath_target[100];
+	char ns_self[100], ns_target[100];
+	ssize_t n_self, n_target;
+
+	sprintf(nspath_self, "/proc/self/ns/%s", ns);
+	n_self = readlink(nspath_self, ns_self, 100);
+	if (n_self < 0)
+	    return -1;
+
+	sprintf(nspath_target, "/proc/%d/ns/%s", Lp->pid, ns);
+	n_target = readlink(nspath_target, ns_target, 100);
+	if (n_target < 0)
+	    return -1;
+
+	if (n_self != n_target)
+	    return -1;
+
+	if (strncmp(ns_self, ns_target, n_self))
+	    return -1;
+
+	return 0;
+}
+
+int
+compare_kernel_release(release)
+	char *release;          /* kernel release */
+{
+	struct utsname uts;
+
+	uname(&uts);
+	return strcmp(uts.release, release);
```

Comment:
strcmp() may not give correct results for versions, e.g.: `3.8.0` > `3.11.0`

Reply:
Would use LINUX_VERSION_CODE instead.